server/commands/list_objects.go

```diff
@@ -238,26 +238,27 @@ func (q *ListObjectsQuery) performChecks(
 	errChan chan<- error,
 	resolvedChan chan<- struct{},
 ) {
-	g := new(errgroup.Group)
-	g.SetLimit(maximumConcurrentChecks)
-	var objectsFound = new(uint32)
-
-	iter1 := storage.NewTupleKeyObjectIterator(req.GetContextualTuples().GetTupleKeys())
+	iter1 := storage.NewTupleKeyObjectIterator(
+		req.GetContextualTuples().GetTupleKeys(),
+		func(tk *openfgapb.TupleKey) bool {
+			return tuple.GetType(tk.GetObject()) == req.GetType()
+		})
 
 	iter2, err := q.Datastore.ListObjectsByType(ctx, req.GetStoreId(), req.GetType())
 	if err != nil {
+		iter1.Stop()
```

Comment:
Should this be `defer`ed after its definition?

Reply:
I don't think so - under normal conditions, the iterator will be stopped in line 256. This is just for the error case, where I need to clean up manually

Reply:
Ah, I see. Thanks!